worker/src/checks/hardware/gpu.rs

```diff
@@ -71,30 +66,50 @@ fn get_gpu_status() -> GpuDevice {
     // Get device count
     let device_count = match nvml.device_count() {
         Ok(count) => count as usize,
-        Err(e) => return GpuDevice::NotAvailable(format!("Failed to get device count: {}", e)),
+        Err(e) => {
+            Console::user_error(&format!("Failed to get device count: {}", e));
+            return vec![];
+        }
     };
 
     if device_count == 0 {
-        return GpuDevice::NotAvailable("No GPU devices detected".to_string());
+        Console::user_error("No GPU devices detected");
+        return vec![];
     }
 
-    // Get first device info
-    // TODO: Get all devices
-    match nvml.device_by_index(0) {
-        Ok(device) => {
-            let name = device.name().unwrap_or_else(|_| "Unknown".to_string());
-            let memory = device.memory_info().map(|m| m.total).unwrap_or(0);
-            let driver_version = nvml
-                .sys_driver_version()
-                .unwrap_or_else(|_| "Unknown".to_string());
+    let mut device_map: std::collections::HashMap<String, GpuDevice> =
+        std::collections::HashMap::new();
+
+    for i in 0..device_count {
+        match nvml.device_by_index(i as u32) {
+            Ok(device) => {
+                let name = device.name().unwrap_or_else(|_| "Unknown".to_string());
+                let memory = device.memory_info().map(|m| m.total).unwrap_or(0);
+                let driver_version = nvml
+                    .sys_driver_version()
+                    .unwrap_or_else(|_| "Unknown".to_string());
 
-            GpuDevice::Available {
-                name,
-                memory,
-                driver_version,
-                device_count,
+                if let Some(existing_device) = device_map.get_mut(&name) {
+                    existing_device.count += 1;
+                    existing_device.indices.push(i as u32);
+                } else {
+                    device_map.insert(
+                        name.clone(),
+                        GpuDevice {
+                            name,
+                            memory,
+                            driver_version,
+                            count: 1,
+                            indices: vec![i as u32],
+                        },
+                    );
+                }
```

Comment:
I looked up a few SMI examples and it seems like the "name" field is usually unique to a product, but also this doesn't seem to be guaranteed, e.g. with VRAM modded cards (that might get more popular in the future) it could cause errors not to check the VRAM available. It might be worthwhile appending the memory as a string to the name and using that as a key.